spec/integration/devtools/rake/flay/verify_spec.rb

```diff
@@ -0,0 +1,171 @@
+require 'spec_helper'
+require 'tempfile'
+require 'tmpdir'
+require 'flay'
+
+# rubocop:disable ModuleLength
+module Devtools
+  describe Rake::Flay, '#verify' do
+    let(:tempfile) { Tempfile.new(%w[file .rb], Dir.mktmpdir) }
+    let(:file) { Pathname(tempfile.path) }
+    let(:directories) { [file.dirname.to_s] }
+    let(:ruby) do
+      <<-ERUBY
+      def foo; end
+      def bar; end
+      ERUBY
+    end
+
+    before do
+      # silence other flay output
+      $stdout = $stderr = StringIO.new
+
+      tempfile.write(ruby)
+      tempfile.close
+    end
+
+    after do
+      # reset out
+      $stdout = STDOUT
+      $stderr = STDERR
+
+      file.unlink
+    end
```

Comment:
IMO its nicer to combine the before and around hook when they have inverse effects into an `around` block.

Also lets do the inverse effect inside a `ensure` block to make sure stderr/stdout are always reset.